RealmSwift/Sync.swift

```diff
@@ -482,6 +504,28 @@ extension SyncUser {
             callback(token, nil)
         }
     }
+
+    /**
+     Create a sync configuration instance.
+
+     Additional settings can be optionally specified. Descriptions of these
+     settings follow.
+     
+     `enableSSLValidation` is true by default. It can be disabled for debugging
+     purposes.
+     
+     - warning: The URL must be absolute (e.g. `realms://example.com/~/foo`), and cannot end with
+     `.realm`, `.realm.lock` or `.realm.management`.
+     
+     - warning: NEVER disable SSL validation for a system running in production.
+     */
+    public func createConfiguration(realmURL: URL, fullSynchronization: Bool = false, enableSSLValidation: Bool = true, urlPrefix: String? = nil) -> Realm.Configuration {
```

Comment:
This should just be named `configuration` and have URL be optional rather than a separate `defaultConfiguration` function.

Returning a `SyncConfiguration` struct works better with how configurations are currently constructed.

Reply:
Huh, I thought `automatic()` returned a `SyncConfiguration`, but on second thought of course it doesn't. Given that, I guess it's correct for this to return a `Realm.Configuration` as well.